test/test_cuda.py

```diff
@@ -4702,6 +4703,92 @@ class TestCudaOptims(TestCase):
     # These tests will be instantiate with instantiate_device_type_tests
     # to apply the new OptimizerInfo structure.
 
+    @onlyCUDA
+    @unittest.skipIf(
+        not TEST_CUDA_GRAPH, "CUDA >= 11.0 or ROCM >= 5.3 required for graphs"
+    )
+    @optims(
+        [
+            optim for optim in optim_db if "fused" in optim.supported_impls
+        ],
+        dtypes=[torch.float32],
+    )
+    def test_grad_scaling_autocast_fused_optimizers(
+        self, device, dtype, optim_info
+    ):
+        # This ut is from test_cuda.py test_grad_scaling_autocast_fused_optimizers
+        # but only test Adam/AdamW on CPU
+        # TODO: haozhe, support SGD and unified this ut with the CUDA only one
+        if device not in optim_info.supports_fused_on:
+            self.skipTest(
+                f"{device} is not supported for fused on {optim_info.optim_cls.__name__}"
+            )
+        optim_inputs = optim_info.optim_inputs_func(device=device)
+        optim_cls = optim_info.optim_cls
+        for optim_input in optim_inputs:
+            for _separate_unscale in (True, False):
+                kwargs = optim_input.kwargs
+                # kwargs["fused"] = False
```

Comment:
had to comment this out for below error. 

https://github.com/pytorch/pytorch/blob/main/torch/optim/adagrad.py#L69

```
RuntimeError: `fused=True` requires all the params to be floating point Tensors of supported devices: ['xpu', 'cpu', 'privateuseone'].
```

Reply:
Well, this is actually the whole point of the test, so we cannot comment it out.

What we should do is remove the onlyCUDA, and still keep this skip:
```
   if device not in optim_info.supports_fused_on:
            self.skipTest(
                f"{device} is not supported for fused on {optim_info.optim_cls.__name__}"
            )
```

Reply:
@janeyx99  that worked, but one test is failing at https://github.com/pytorch/pytorch/blob/main/torch/optim/adagrad.py#L540 with 
```   File "/home/sandish/pytorch/torch/optim/adagrad.py", line 540, in _fused_adagrad
    torch._fused_adagrad_(
  File "/home/sandish/pytorch/torch/__init__.py", line 2023, in __getattr__
    raise AttributeError(f"module '{__name__}' has no attribute '{name}'")
AttributeError: module 'torch' has no attribute '_fused_adagrad_'
```

Reply:
that might be ur local build being outdated

Reply:
there is a lint failure though, please run lintrunner before committing

Reply:
the local lintrunner shows opposite of the PR linterunner. 

```
    You can run `lintrunner -a` to apply this patch.

    964  964 |         # SparseAdam raises a warning and not an error for the first entry. We
    965  965 |         # update it here:
    966  966 |         error_inputs[0].error_type = FutureWarning
    966      |-        error_inputs[
    967      |-            0
    968      |-        ].error_regex = "Passing in a raw Tensor as ``params`` to SparseAdam"
         967 |+        error_inputs[0].error_regex = (
         968 |+            "Passing in a raw Tensor as ``params`` to SparseAdam"
         969 |+        )
    970  970 | 
    971  971 |         error_inputs += [
    972  972 |             ErrorOptimizerInput(
```
but fixed as per PR lintrunner